projects/driver_controls/test/test_throttle.c

```diff
@@ -0,0 +1,91 @@
+#include <string.h>
+#include "ads1015.h"
+#include "ads1015_def.h"
+#include "delay.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "throttle.h"
+#include "unity.h"
+
+static Ads1015Storage ads1015_storage;
+static ThrottleStorage throttle_storage;
+static ThrottleCalibrationData calibration_data;
+
+// This will be way of setting calibration data before the calibration routine is implemented.
+static void prv_set_calibration_data(ThrottleCalibrationData *data, int16_t min, int16_t brake_max,
+                                     int16_t coast_max, int16_t accel_max, int16_t tolerance) {
+  data->zone_thresholds_main[THROTTLE_ZONE_BRAKE][THROTTLE_THRESH_MIN] = min * 2;
+  data->zone_thresholds_main[THROTTLE_ZONE_BRAKE][THROTTLE_THRESH_MAX] = brake_max * 2;
+  data->zone_thresholds_main[THROTTLE_ZONE_COAST][THROTTLE_THRESH_MIN] = brake_max * 2;
+  data->zone_thresholds_main[THROTTLE_ZONE_COAST][THROTTLE_THRESH_MAX] = coast_max * 2;
+  data->zone_thresholds_main[THROTTLE_ZONE_ACCEL][THROTTLE_THRESH_MIN] = coast_max * 2;
+  data->zone_thresholds_main[THROTTLE_ZONE_ACCEL][THROTTLE_THRESH_MAX] = accel_max * 2;
+  data->zone_thresholds_main[THROTTLE_ZONE_ALL][THROTTLE_THRESH_MIN] = min * 2;
+  data->zone_thresholds_main[THROTTLE_ZONE_ALL][THROTTLE_THRESH_MAX] = accel_max * 2;
+  data->zone_thresholds_secondary[THROTTLE_ZONE_BRAKE][THROTTLE_THRESH_MIN] = min;
+  data->zone_thresholds_secondary[THROTTLE_ZONE_BRAKE][THROTTLE_THRESH_MAX] = brake_max;
+  data->zone_thresholds_secondary[THROTTLE_ZONE_COAST][THROTTLE_THRESH_MIN] = brake_max;
+  data->zone_thresholds_secondary[THROTTLE_ZONE_COAST][THROTTLE_THRESH_MAX] = coast_max;
+  data->zone_thresholds_secondary[THROTTLE_ZONE_ACCEL][THROTTLE_THRESH_MIN] = coast_max;
+  data->zone_thresholds_secondary[THROTTLE_ZONE_ACCEL][THROTTLE_THRESH_MAX] = accel_max;
+  data->zone_thresholds_secondary[THROTTLE_ZONE_ALL][THROTTLE_THRESH_MIN] = min;
+  data->zone_thresholds_secondary[THROTTLE_ZONE_ALL][THROTTLE_THRESH_MAX] = accel_max;
+  data->channel_readings_tolerance = tolerance;
+}
+
+void setup_test(void) {
+  gpio_init();
+  interrupt_init();
+  gpio_it_init();
+  soft_timer_init();
+  I2CSettings i2c_settings = {
+    .speed = I2C_SPEED_FAST,                    //
+    .scl = { .port = GPIO_PORT_B, .pin = 10 },  //
+    .sda = { .port = GPIO_PORT_B, .pin = 11 },  //
+  };
+  i2c_init(TEST_ADS1015_I2C_PORT, &i2c_settings);
+  GPIOAddress ready_pin = {
+    .port = GPIO_PORT_B,  //
+    .pin = 2,             //
+  };
+  ads1015_init(&ads1015_storage, TEST_ADS1015_I2C_PORT, TEST_ADS1015_ADDR, &ready_pin);
+  prv_set_calibration_data(&calibration_data, 0, 25, 50, 75, 0);
```

Comment:
You have some values you could use from the actual pedal. Why don't you just use that?

Reply:
I used these ones because ADS1015 on x86 always gives 0 and that's not even within the bounds of my values. Even if I change that reading to some other number I still can't verify the behavior so much since I have a constant reading (following up with your next comment.)

Reply:
You should consider a way to do so. Testing behavior is much more important than validating arguments.